freight/providers/utils.py

```diff
@@ -12,11 +10,16 @@ def parse_provider_config(type, config):
         raise ApiError(message=f"Invalid provider: {type}", name="invalid_provider")
 
     result = {}
-    all_options = chain(
-        list(instance.get_default_options().items()),
-        list(instance.get_options().items()),
-    )
-    for option, option_values in all_options:
+    all_options = {**instance.get_default_options(), **instance.get_options()}
+
+    for key in config:
+        if not all_options.get(key):
```

Comment:
`if key not in all_options` perhaps? -- that way if it's `False` or falsey by default it'll not produce an error?